cmd/main.go

```diff
@@ -138,3 +150,50 @@ func getWatchNamespace() (string, error) {
 	}
 	return ns, nil
 }
+
+// GetEnvVar returns the value from an environment variable
+// or the provided default value if the variable does not exist.
+func getEnvVarString(varName, defaultValue string) string {
+	if value, exists := os.LookupEnv(varName); exists {
+		return value
+	}
+	return defaultValue
+}
+
+// Returns time.Duration parsed from an env variable
+// or the default value if the variable does not exist or does not parse into a duration.
+func getEnvVarDuration(varName string, defaultValue time.Duration) time.Duration {
+	val := getEnvVarString(varName, "")
+
+	if val == "" {
+		return defaultValue
+	}
+
+	parsedVal, err := time.ParseDuration(val)
+	if err != nil {
+		setupLog.Error(err, "Invalid boolean format for environment variable", "Variable", varName, "Value", val)
+		return defaultValue
+	}
+	return parsedVal
+
+}
+
+// validateLeaderElectionConfig returns the Namespace the operator should be watching for changes
+func validateLeaderElectionConfig(leaseDuration, renewDeadline, retryPeriod time.Duration) (time.Duration, time.Duration, time.Duration) {
+
+	// validate that renewDeadlne < leaseDuration
+	if renewDeadline >= leaseDuration {
+		setupLog.Info("Invalid configuration: LEADER_ELECTION_RENEW_DEADLINE must be less that LEADER_ELECTION_LEASE_DURATION; using default values")
```

Comment:
might be worth adding to these error messages what the values are being overridden to

Reply:
good call 👍🏾, especially for future troubleshooting efforts